CHANGELOG.rst

```diff
@@ -10,51 +10,50 @@ TBR
 
 * In order for the above fix to be possible, overriding the callback dependencies
   created by **scrapy-poet** via :attr:`scrapy.http.Request.cb_kwargs` is now
-  unsupported. This is a backward incompatible change.
+  unsupported. This is a **backward incompatible** change.
+
+* Improved behavior when ``spider.parse()`` method arguments are supposed
+  to be provided by **scrapy-poet**. Previously, it was causing
+  unnecessary work in unexpected places like
+  :class:`scrapy.downloadermiddlewares.robotstxt.RobotsTxtMiddleware`,
+  :class:`scrapy.pipelines.images.ImagesPipeline` or
+  :class:`scrapy.pipelines.files.FilesPipeline`. It is also a reason
+  :class:`web_poet.page_inputs.client.HttpClient` might not be working
+  in page objects. Now these cases are detected, and a warning is issued.
+
+  As of Scrapy 2.7, it is not possible to fix the issue completely
+  in scrapy-poet. Fixing it would require Scrapy changes; some 3rd party
+  libraries may also need to be updated.
+
+  .. note::
+
+      The root of the issue is that when request.callback is None, ``parse``
+      callback is assumed normally. But sometimes callback=None is used
+      when :class:`scrapy.http.Request` is added to the Scrapy's downloader
+      directly, in which case no callback is used. Middlewares, including
+      **scrapy-poet**'s, can't distinguish between these two cases, which causes
+      all kinds of issues.
+
+  We recommend all **scrapy-poet** users to modify their code to
+  avoid the issue. Please **don't** define ``parse``
```

Comment:
```suggestion
  avoid the issue. Please **don't** define ``parse()``
```

There are some unchanged lines that has this style. We could go with either ``parse()`` or ``parse`` but we need to make them consistent throughout the changelog.

Reply:
Makes sense. Fixed in https://github.com/scrapinghub/scrapy-poet/pull/114/commits/123aa340e36690643d722a08edf34f90a62ee2b1